service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/bqdataset/ControlledBigQueryDatasetResource.java

```diff
@@ -241,7 +241,8 @@ public ControlledBigQueryDatasetResource.Builder datasetName(String datasetName)
         // But if the resource name is not a valid dataset name, we will need to generate a unique
         // dataset id.
         ResourceValidationUtils.validateBqDatasetName(datasetName);
-        this.datasetName = datasetName;
+        this.datasetName =
+            ControlledBigQueryDatasetHandler.getHandler().generateCloudName(null, datasetName);
       } catch (InvalidReferenceException e) {
         this.datasetName = generateUniqueDatasetId();
```

Comment:
it looks like this will fall back to generating a random UUID-based dataset name if the dataset name doesn't work. If `generateCloudName` is guaranteed to return a usable cloud name, can you just call that instead?